src/main/java/com/aws/greengrass/mqttclient/MqttClient.java

```
@@ -493,43 +493,38 @@ public CompletableFuture<SubscribeResponse> subscribe(Subscribe request)
+                Consumer<Publish> cb = (Publish m) -> request.getCallback().accept(new MqttMessage(m.getTopic(),
+                        m.getPayload(), QualityOfService.getEnumValueFromInteger(m.getQos().getValue()), m.isRetain()));
 
                 return Subscribe.builder().qos(QOS.fromInt(request.getQos().getValue())).topic(request.getTopic())
                         .callback(cb).build();
             });
 
-            subscribe(newReq)
-                    .thenApply((v) -> {
-                        // null is a success because subscribe returns null if the subscription already existed
-                        if (v == null || v.isSuccessful()) {
-                            return v;
-                        }
-                        String rcString = SubAckPacket.SubAckReasonCode.UNSPECIFIED_ERROR.name();
-                        try {
-                            rcString = SubAckPacket.SubAckReasonCode.getEnumValueFromInteger(v.getReasonCode()).name();
-                        } catch (RuntimeException ignored) {
-                        }
-                        // Consumers of this deprecated API expect to receive an MqttException if subscribing fails
-                        throw new MqttException(
-                                "Error subscribing. Reason: " + rcString);
-                    })
-                    .get(getMqttOperationTimeoutMillis(), TimeUnit.MILLISECONDS);
+            subscribe(newReq).thenApply((v) -> {
+                // null is a success because subscribe returns null if the subscription already existed
+                if (v == null || v.isSuccessful()) {
+                    return v;
+                }
+                String rcString = SubAckPacket.SubAckReasonCode.UNSPECIFIED_ERROR.name();
+                try {
+                    rcString = SubAckPacket.SubAckReasonCode.getEnumValueFromInteger(v.getReasonCode()).name();
+                } catch (RuntimeException ignored) {
```

Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Swallowing exceptions without logging can make it difficult to identify and troubleshoot issues, potentially leading to application instability, hidden bugs, or unexpected behavior. To avoid this, ensure that all exceptions are logged using appropriate logging levels. When logging exceptions, include the exception object or its stack trace to capture detailed information. For critical exceptions, rethrow them when necessary to provide sufficient context for effective troubleshooting.

Suggested remediation:
Handle all exceptions by creating a log for each exception using the message thrown.
```
@@ -523,2 +523,3 @@
                 } catch (RuntimeException ignored) {
+                    log.error(&#34;RuntimeException is caught.&#34;, ignored);
                 }

```